eth/catalyst/api_test.go

```diff
@@ -874,7 +875,7 @@ func TestNewPayloadOnInvalidTerminalBlock(t *testing.T) {
 	n, ethservice := startEthService(t, genesis, preMergeBlocks)
 	defer n.Close()
 
-	genesis.Config.TerminalTotalDifficulty = preMergeBlocks[0].Difficulty() //.Sub(genesis.Config.TerminalTotalDifficulty, preMergeBlocks[len(preMergeBlocks)-1].Difficulty())
+	ethservice.BlockChain().Config().TerminalTotalDifficulty = preMergeBlocks[len(preMergeBlocks)-1].Difficulty() //.Sub(genesis.Config.TerminalTotalDifficulty, preMergeBlocks[len(preMergeBlocks)-1].Difficulty())
```

Comment:
How is this correct? You are setting the `TTD` to the `D` of the last block, right? That is correct IFF the last block is the entire chain.

Reply:
So, there are two changes here. 
1. Changing from direct accessor via `genesis.Config.TerminalTotaDifficulty` to accessign via `Config()`. 
2. Modifying to use the last premergeBlock instead of the first. 

Change 1 seems ok to me, and might avoid a race. 
Change 2 seems like an arbitrary change, just as incorrect after as it was before.

Did I misunderstand something?